File: 백준/Silver/1920. 수 찾기/수 찾기.swift
let n = Int(readLine() ?? "") ?? 0
let a = Set((readLine() ?? "").split { $0 == " " }.map { Int($0) ?? 0 })
let m = Int(readLine() ?? "") ?? 0
let b = (readLine() ?? "").split { $0 == " " }.map { Int($0) ?? 0 }

b.forEach {
    print(a.contains($0) ? "1" : "0")
}

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        Set<String> set = new HashSet(Arrays.asList(br.readLine().split(" ")));
        
        int m = Integer.parseInt(br.readLine());

        StringBuilder sb = new StringBuilder();
        for (String num : br.readLine().split(" ")) {
            sb.append(set.contains(num) ? "1" : "0").append("\n");
        }

        System.out.println(sb);
    }
}